frontend/lib/js/category-trees/reducer.js

```diff
@@ -10,24 +10,93 @@ import {
   LOAD_TREE_SUCCESS,
   LOAD_TREE_ERROR,
   CLEAR_TREES,
-}                       from './actionTypes';
-
-import { setTree }      from './globalTreeStoreHelper';
+  SEARCH_TREES_START,
+  SEARCH_TREES_END,
+  SEARCH_TREES_ERROR,
+}                                         from './actionTypes';
+import { setTree }                        from './globalTreeStoreHelper';
 
 export type TreesType = { [treeId: string]: NodeType }
 
+export type SearchType = {
+  searching: boolean,
+  loading: boolean,
+  query: string,
+  words: Array<string>,
+  result: Array<TreeNodeIdType>,
+  limit: number,
+  resultCount: number,
+  duration: number
+}
+
 export type StateType = {
   loading: boolean,
   version: any,
-  trees: TreesType
+  trees: TreesType,
+  search?: SearchType,
 };
 
 const initialState: StateType = {
   loading: false,
   version: null,
-  trees: {}
+  trees: {},
+  search: {
+    searching: false,
+    loading: false,
+    query: '',
+    words: [],
+    result: [],
+    limit: 0,
+    resultCount: 0,
+    duration: 0
+  }
 };
 
+const setSearchTreesEnd = (state: StateType, action: Object): StateType => {
+  const { query, result } = action.payload;
+  const searching = query && query.length > 0;
+  const maxResult = [];
+  const limit = parseInt(process.env.SEARCH_RESULT_LIMIT);
+
+  for (var i = 0; i < result.length; i++) {
+    maxResult.push(result[i]);
```

Comment:
Wouldn't it make sense to only return so many results from the server? We could add a query parameter to the search request to send the current `SEARCH_RESULT_LIMIT`. Then, the server would need to supply the number of total results as wel...

Reply:
yes, agree. That was also my consideration but then we need a new display of the result set or this can be removed.